### ISSUE TITLE
Typescript definition missing createIndex

### ISSUE THREAD

**Issue by crystalfp** [apache/couchdb-nano#169]
## Expected Behavior
```
import Nano from "nano";
const nano = Nano("http://localhost:5984");

const mydb = nano.db.use("mydb");
const indexDef = {index: {fields: ["tag"]}, name: "tagIndex"};
mydb.createIndex(indexDef).then((result: {[key: string]: string}) => console.log(result));
```
This is valid Typescript.

## Current Behavior
During compilation Typescript complains that:
`Property 'createIndex' does not exist on type 'DocumentScope<any>'.ts(2339)`

## Possible Solution
In `nano.d.ts` add inside `interface DocumentScope` a definition along the lines of:
```
createIndex(opt: {index: {fields: string[]}; name: string;}): Promise<any>;
```

## Steps to Reproduce (for bugs)

1. Put the code in a file
2. Provide the usual tsconfig.json
3. Compile with `tsc`
4. See the error

## Context
It is an annoying error from typescript.

## Your Environment

* Version used:   "nano": "^8.1.0",
* Browser Name and version: Chrome latest
* Operating System and version (desktop or mobile): Win 10 64bit laptop
* Link to your project: private for now

**Comment by glynnbird** [comment]
Looking at this issue, I note that `createIndex` [is defined in the TypeScript definitions](https://github.com/apache/couchdb-nano/issues/169) and I can use it without any complaint from the `tsc` compiler:

```
const nano = Nano("http://localhost:5984");
const mydb = nano.db.use("mydb");
const indexDef = {index: {fields: ["tag"]}, name: "tagIndex"};
mydb.createIndex(indexDef).then((d) => {
  console.log(d.id)
})
```

**Comment by crystalfp** [comment]
Well, the copy installed with nano 8.1.0 is different (see attached). Which version is your? I tried to disinstall and reinstall nano but nothing changes. The definition file is the one I had before that is attached below.

[nano-old.d.ts.txt](https://github.com/apache/couchdb-nano/files/3359895/nano-old.d.ts.txt)

Confused.

**Comment by crystalfp** [comment]
Seems your definition is "in-transit" between GitHub, DefinitelyTyped and npm package. Maybe I should only wait for the new nano package release. In the meantime I copied your definition in node_modules/nano/lib (bad bad bad!, but at least I silenced tsc.

**Comment by crystalfp** [comment]
Spoke too fast! The compilation complains about parameters with implicit any type. Here are the compiler's errors that can be fixed before the detinition file lands in the new package version.

```
PS D:\Projects\CouchDB> tsc
node_modules/nano/lib/nano.d.ts:63:16 - error TS7006: Parameter 'source' implicitly has an 'any' type.
63         enable(source, target, opts0, callback0?): any;
                  ~~~~~~

node_modules/nano/lib/nano.d.ts:63:24 - error TS7006: Parameter 'target' implicitly has an 'any' type.
63         enable(source, target, opts0, callback0?): any;
                          ~~~~~~

node_modules/nano/lib/nano.d.ts:63:32 - error TS7006: Parameter 'opts0' implicitly has an 'any' type.
63         enable(source, target, opts0, callback0?): any;
                                  ~~~~~

node_modules/nano/lib/nano.d.ts:63:39 - error TS7006: Parameter 'callback0' implicitly has an 'any' type.
63         enable(source, target, opts0, callback0?): any;
                                         ~~~~~~~~~~

node_modules/nano/lib/nano.d.ts:64:17 - error TS7006: Parameter 'id' implicitly has an 'any' type.
64         disable(id, rev, opts0, callback0?): any;
                   ~~

node_modules/nano/lib/nano.d.ts:64:21 - error TS7006: Parameter 'rev' implicitly has an 'any' type.
64         disable(id, rev, opts0, callback0?): any;
                       ~~~

node_modules/nano/lib/nano.d.ts:64:26 - error TS7006: Parameter 'opts0' implicitly has an 'any' type.
64         disable(id, rev, opts0, callback0?): any;
                            ~~~~~

node_modules/nano/lib/nano.d.ts:64:33 - error TS7006: Parameter 'callback0' implicitly has an 'any' type.
64         disable(id, rev, opts0, callback0?): any;
                                   ~~~~~~~~~~

node_modules/nano/lib/nano.d.ts:65:15 - error TS7006: Parameter 'id' implicitly has an 'any' type.
65         query(id, opts0, callback0?): any;
                 ~~

node_modules/nano/lib/nano.d.ts:65:19 - error TS7006: Parameter 'opts0' implicitly has an 'any' type.
65         query(id, opts0, callback0?): any;
                     ~~~~~

node_modules/nano/lib/nano.d.ts:65:26 - error TS7006: Parameter 'callback0' implicitly has an 'any' type.
65         query(id, opts0, callback0?): any;
                            ~~~~~~~~~~
```

**Comment by crystalfp** [comment]
Problem seem solved in nano 9.0.1
Closing and thanks!
mario
